jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -2307,37 +2307,55 @@ public boolean equals(Object object) {
 		return false;
 	}
 
-	private static final void deduplicateStrings(String s1, String s2) {
-		if ((s1.count ^ s2.count) >= 0) {
+	/**
+	 * Deduplicate the backing buffers of the given strings.
+	 *
+	 * This updates the {@link #value} of one of the two given strings so that
+	 * they both share a single backing buffer. The strings must have identical
+	 * contents.
+	 *
+	 * Deduplication helps save space, and lets {@link #equals(Object)} exit
+	 * early more often.
+	 *
+	 * The strings' corresponding backing buffers are accepted as parameters
+	 * because the caller likely already has them.
+	 *
+	 * @param s1
+	 *			  the first string
+	 * @param value1
+	 *			  {@code s1.value}
+	 * @param s2
+	 *			  the second string
+	 * @param value2
+	 *			  {@code s2.value}
+	 */
```

Comment:
Same comment about formatting here.  The javadoc is inconsistently formatted throughout the OpenJ9 class libraries.  New javadoc should adhere to the formats that are commonly used.  Over time this will hopefully lead to cleaner formatting.